src/firefly/js/drawingLayers/WebGridUI.jsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ * Lijun
+ * 4/15/16
+ */
+import React, {PropTypes} from 'react';
+import {dispatchForceDrawLayerUpdate,dispatchWebGridLayerCoordinateChange} from '../visualize/DrawLayerCntlr.js';
+import AppDataCntlr from '../core/AppDataCntlr.js';
+import {primePlot,getPlotViewById} from '../visualize/PlotViewUtil.js';
+import {SingleColumnMenu} from '../ui/DropDownMenu.jsx';
+import CoordinateSys from '../visualize/CoordSys.js';
+import lodash, {get} from 'lodash';
+import {ListBoxInputFieldView} from '../ui/ListBoxInputField.jsx';
+import {COORDIANTE_PREFERENCE} from './WebGrid.js';
+import {
+    ToolbarButton,
+    DropDownVerticalSeparator} from '../ui/ToolbarButton.jsx';
+import {DropDownToolbarButton} from '../ui/DropDownToolbarButton.jsx';
+
+export const getUIComponent = (drawLayer,pv) => < WebGridUI drawLayer={drawLayer} pv={pv}/>;
+
+const coordinateOptionArray = [
+    {label:'Eq. J2000',         value:'eq2000hms'},
+    {label: 'Eq. J2000 Decimal',value:'eq2000dcm'},
+    {label: 'Eq. B1950',        value:'eqb1950hms'},
+    {label:'Eq. B1950 Decimal', value:'eqb1950dcm'},
+    {label:'Galactic',          value:'galactic'},
+    {label: 'Super Galactic',   value:'superGalactic'},
+    {label:'Ecliptic J2000',    value:'epj2000'},
+    {label:'Ecliptic B1950',    value:'epb1950'}
+
+];
+
+
+function WebGridUI({drawLayer,pv}) {
+
+    var pref= AppDataCntlr.getPreference(COORDIANTE_PREFERENCE);
+   return  (
```

Comment:
Using the preference is a good idea but you should keep the value in the drawing layer too.  Put it in the drawing layer with `dispatchModifyCustomField` (you will have to watch this in the drawing layer as well). Put it in the preference as well but just use the preference if it is not in the drawing layer. If you do it like this, you will not have to use `dispatchForceDrawLayerUpdate`.